sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageSerializer.java

```diff
@@ -169,8 +175,15 @@ public int getSize(org.apache.qpid.proton.message.Message amqpMessage) {
         Objects.requireNonNull(message, "'message' cannot be null.");
         Objects.requireNonNull(clazz, "'clazz' cannot be null.");
 
+        ExtendedAmqpMessage extendedAmqpMessage =  null;
+
+        if (message instanceof ExtendedAmqpMessage) {
+            extendedAmqpMessage = (ExtendedAmqpMessage) message;
+        }
         if (clazz == ServiceBusReceivedMessage.class) {
-            return (T) deserializeMessage(message);
+            ServiceBusReceivedMessage brokeredMessage = deserializeMessage(message);
+            brokeredMessage.setLockToken(MessageUtils.convertDotNetBytesToUUID(extendedAmqpMessage.getDeliveryTag().array()));
```

Comment:
Does `ExtendedAmqpMessage`'s delivery tag property ever get used as a `ByteBuffer`? Could it get stored as a `byte[]` instead?